packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -24,9 +24,28 @@ import 'package:flutter/scheduler.dart';
 
 import 'basic.dart';
 import 'framework.dart';
+import 'layout_builder.dart';
 import 'lookup_boundary.dart';
 import 'ticker_provider.dart';
 
+/// The signature of the widget builder callback used in
+/// [OverlayPortal.overlayChildLayoutBuilder].
+///
+/// {@template flutter.widgets.OverlayChildLayoutBuilder}
+/// The `childSize` parameter and the `overlaySize` parameter are the size of
+/// [OverlayPortal.child] and [Overlay] respectively, in their own coordinates.
+///
+/// The `childPaintTransform` parameter is the paint transform of
+/// [OverlayPortal.child], in the target [Overlay]'s coordinates.
+/// {@endtemplate}
+typedef OverlayChildLayoutBuilder =
+    Widget Function(
+      BuildContext context,
+      Size childSize,
+      Matrix4 childPaintTransform,
+      Size overlaySize,
```

Comment:
Instead of passing in individual parameters, should we pass in an immutable object that gives you access to these values? Would make it easier in the future to extend this with more values.